homeassistant/components/slide_local/config_flow.py

```diff
@@ -0,0 +1,152 @@
+
+import logging
+from typing import Any
+
+from goslideapi.goslideapi import (
+    AuthenticationFailed,
+    ClientConnectionError,
+    ClientTimeoutError,
+    DigestAuthCalcError,
+    GoSlideLocal as SlideLocalApi,
+)
+import voluptuous as vol
+
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_API_VERSION, CONF_HOST, CONF_MAC, CONF_PASSWORD
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.selector import (
+    BooleanSelector,
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import CONF_INVERT_POSITION, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+BOOLEAN_SELECTOR = BooleanSelector()
+
+API_VERSION_SELECTOR = SelectSelector(
+    SelectSelectorConfig(
+        options=[
+            SelectOptionDict(value="1", label="API 1"),
+            SelectOptionDict(value="2", label="API 2"),
+        ],
+        mode=SelectSelectorMode.DROPDOWN,
+        translation_key=CONF_API_VERSION,
+    )
+)
+
+
+class SlideConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for slide_local."""
+
+    _mac: str = ""
+    _host: str = ""
+
+    VERSION = 1
+    MINOR_VERSION = 1
+
+    async def async_test_connection(
+        self, user_input: dict[str, str | int]
+    ) -> dict[str, str]:
+        """Reusable Auth Helper."""
+        slide = SlideLocalApi()
+        await slide.slide_add(
+            user_input[CONF_HOST],
+            user_input.get(CONF_PASSWORD, ""),
+            user_input[CONF_API_VERSION],
+        )
+
+        try:
+            result = await slide.slide_info(user_input[CONF_HOST])
+            self._mac = result["mac"]
+        except (ClientConnectionError, ClientTimeoutError):
+            return {"base": "cannot_connect"}
+        except (AuthenticationFailed, DigestAuthCalcError):
+            return {"base": "invalid_auth"}
+        except Exception:  # noqa: BLE001
+            _LOGGER.exception("Exception occurred during connection test")
+            return {"base": "unknown"}
+
+        return {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the user step."""
+        errors = {}
+        if user_input is not None:
+            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+            user_input[CONF_API_VERSION] = int(user_input[CONF_API_VERSION])
+
+            if not (errors := await self.async_test_connection(user_input)):
+                await self.async_set_unique_id(self._mac)
+                user_input |= {CONF_MAC: format_mac(self._mac)}
+
+                return self.async_create_entry(
+                    title=user_input[CONF_HOST], data=user_input
+                )
+
+        if user_input is not None and user_input.get(CONF_HOST) is not None:
```

Comment:
This happens, if user provided information that did not pass the connection test (e.g. wrong password). I thought it would be nice to pre-fill the host, as most likely it is the password or API version that needs to be changed.